protocols/gossipsub/src/types.rs

```diff
@@ -99,7 +100,7 @@ pub struct RawMessage {
     pub source: Option<PeerId>,
 
     /// Content of the message. Its meaning is out of scope of this library.
-    pub data: Vec<u8>,
+    pub data: Bytes,
```

Comment:
This is also public so should be mentioned in the changelog.

Perhaps we can say something like: "Optimize memory usage through use of `Bytes` in various places: ..." and then list all the APIs that were changed.